README.adoc

```diff
@@ -4950,6 +4950,28 @@ LOREM
 "when an unknown printer took a galley of type and scrambled it to make a type specimen book."
 ----
 
+== Enumerable
+
+=== lazy [[lazy]]
+
+Use `Enumerable#lazy` in combination with methods like `.first` or `.take`, especially for operations on large collections or datasets, because it avoids the need to process the entire collection.
+
+[source,ruby]
+----
+SIZE = 10
+MAX = 10_000_000
+
+# bad
```

Comment:
You might also want to mention that this will instantiate a huge collection first, so it's clearer what the problem and value proposition is.